Arm/BitVec.lean

```diff
@@ -672,8 +672,27 @@ protected theorem shift_left_zero_eq (n : Nat) (x : BitVec n) : x <<< 0 = x := b
     intro i
     simp only [toNat_shiftLeft, Nat.shiftLeft_zero, toNat_mod_cancel]
 
+---------------------------- Negate Lemmas ---------------------------
+
+
+@[simp]
+theorem neg_neg (x : BitVec w₁) : - (- x) = x := by
+  apply BitVec.eq_of_toNat_eq
+  simp only [toNat_neg]
+  by_cases h : x.toNat = 0
+  · simp [h]
```

Comment:
I know that this is a terminal `simp`. I'd like to keep it this way, since these are *pure* Lean-related `BitVec` lemmas that I wish to upstream.